Test/dafny0/SubsetTypes.dfy

```diff
@@ -176,9 +176,9 @@ module SoftCasts {
   {
     if
     case true =>
-      g := z => if z < 0 then -z else z;
+      // g := z => if z < 0 then -z else z; // type error
```

Comment:
So then, what would be the correct way to write such an `int -> nat` absolute-value function?

Reply:
Could always put an `as nat` on the expression (after the =>). But we'll see if the verifier can be encouraged to check this.